### ISSUE TITLE
Expose Project sdk format to be edited via extensions

### ISSUE THREAD

**Issue by sunandabalu** [dotnet/upgrade-assistant#614]
## Summary

Customers wishing to write an extension to be able to add to sdk format need the sdk in IProjectFile to be editable.

## Examples

Specific example from a customer is wishing to have sdk formats specified like so in addition to the default `Microsoft.NET.sdk`:

```
<Project Sdk="XYZ.Build.Before;Microsoft.NET.Sdk;XYZ.Build.After">
```

## Risks / unknowns

The `IsSDK` check in IProjectFile determines if a project is converted to sdk-style, the change to have custom sdk project settings should not break this functionality as it is quire prevalently used.

Need a way to extend IComponentIdentifier so we can determine what kind of component the custom sdk is.

**Comment by sunandabalu** [comment]
Change the sdk property in IProjectFile to make it settable, maybe a collection to facilitate adds/removes. This is good to start with.

**Comment by sunandabalu** [comment]
Closed via #642
